src/providers/user.ts

```diff
@@ -0,0 +1,44 @@
+import { Provider, Resource } from "@/types";
+import { api } from "@/lib/Api";
+import { handleAxiosError } from "@/lib/handleAxiosError";
+import { AxiosError } from "axios";
+import { Bearer } from "@/types";
+import { User, ListOfUser, UserPayload } from "@/types/user";
+
+export const UserProvider: Provider<UserPayload[], User, ListOfUser> = {
+  save: async (resource: Resource<UserPayload[]>): Promise<ListOfUser> => {
+    const { token, payload } = resource;
+    try {
+      const response = await api.put<ListOfUser>(
+        "/users",
+        payload,
+        Bearer(token)
+      );
+      return response.data;
+    } catch (error) {
+      handleAxiosError(error as AxiosError);
+    }
+  },
+  findByOther: async (params: { token: string }): Promise<User> => {
+    const { token } = params;
+    try {
+      const response = await api.put<User>("/users/whoami", Bearer(token));
```

Comment:
why are you doing a put request on whoami ?

Reply:
sorry